### PR TITLE
Implemented ticket creation and rendered on dashboard

### PR DESCRIPTION
**Pull request by vijay1172**
Implemented ticket creation functionality and then worked on displaying them on dashboard
WhatsApp Image 2024-06-30 at 19 41 27_01a07a6a
WhatsApp Image 2024-06-30 at 19 39 36_8b2cab6a

### CODE REVIEW

File: src/Components/TicketList.jsx [hunk 1]
```diff
@@ -1,37 +1,40 @@
...
-import './TicketList.css'; // Importing the CSS file for styling
+import './TicketList.css';
+
+// Truncation function
+const truncate = (str, n) => {
+  return (str && str.length > n) ? str.substr(0, n-1) + '...' : str;
+};
 
-// Functional component to display a list of tickets
 const TicketList = ({ tickets }) => {
```
Comment (Jha2508) [inline src/Components/TicketList.jsx:26]: why are we remmoving documentation?

File: src/Components/TicketList.jsx [hunk 2]
```diff
@@ -1,37 +1,40 @@
...
         <thead>
           <tr>
-            <th>Status</th>
             <th>ID</th>
             <th>Subject</th>
-            <th>Requester</th>
-            <th>Requester Updated</th>
-            <th>Group</th>
-            <th>Assignee</th>
+            <th>Description</th>
```
Comment (Jha2508) [inline src/Components/TicketList.jsx]: Fetch all these from constants

File: src/Screens/Dashboard.jsx [hunk 3]
```diff
@@ -50,14 +51,37 @@ const Dashboard = () => {
 
     fetchTickets(); // Call the fetchTickets function
   }, [toggleErrorState]); // Add toggleErrorState to dependency array to avoid ESLint warning
-
+  const refreshTickets = async () => {
```
Comment (Jha2508) [inline src/Screens/Dashboard.jsx]: documentation for the function

File: src/Screens/Dashboard.jsx [hunk 4]
```diff
@@ -50,14 +51,37 @@ const Dashboard = () => {
...
+        const open = result.tickets.filter(ticket => ticket.status === 'open').length;
+        const onHold = result.tickets.filter(ticket => ticket.status === 'onHold').length;
+        const unassigned = result.tickets.filter(ticket => !ticket.assignee).length;
+        
+        setStats({ unresolved, overdue, dueToday, open, onHold, unassigned });
+      } else {
+        toggleErrorState('Failed to fetch tickets. Please try again later.');
+      }
+    } catch (error) {
+      toggleErrorState('An error occurred while fetching tickets.');
```
Comment (Jha2508) [inline src/Screens/Dashboard.jsx]: move to constants and then use these strings

File: src/utils/networkHelper.js [hunk 5]
```diff
@@ -69,72 +71,74 @@ const viewTicketsAccordingToStatus = async (status, toggleErrorState) => {
...
-    const snapshot = await db.collection(collection).get();
-    return snapshot;
+    const collectionRef = collection(db, collectionName);
+    const snapshot = await getDocs(collectionRef);
+    return snapshot.docs.map(doc => ({ id: doc.id, ...doc.data() }));
   } catch (error) {
-    toggleErrorState(`Error fetching documents from collection ${collection}: ${error.message}`,true);
-    throw new Error(`Error fetching documents from collection ${collection}: ${error.message}`);
+    toggleErrorState(`Error fetching documents from collection ${collectionName}: ${error.message}`, true);
+    throw new Error(`Error fetching documents from collection ${collectionName}: ${error.message}`);
```
Comment (Jha2508) [inline src/utils/networkHelper.js]: Why are we throwing Error here ? what is the purpose of catch if we are throwing error even if we know we need to handle the error on ui here?

**Comment by Jha2508** [comment]
Rest seems good

File: src/Components/TicketList.jsx (hunk 1 above)
Comment (vijay1172) [reply src/Components/TicketList.jsx:26]: Done @Jha2508

File: src/Components/TicketList.jsx (hunk 2 above)
Comment (vijay1172) [reply src/Components/TicketList.jsx]: Done @Jha2508

File: src/Screens/Dashboard.jsx (hunk 3 above)
Comment (vijay1172) [reply src/Screens/Dashboard.jsx]: Updated..

File: src/utils/networkHelper.js (hunk 5 above)
Comment (vijay1172) [reply src/utils/networkHelper.js]: Done

File: src/Screens/Dashboard.jsx (hunk 4 above)
Comment (vijay1172) [reply src/Screens/Dashboard.jsx]: Done

**Review by Jha2508 — APPROVED** [review]

**Pull request merged**